src/divider.jsx

```diff
@@ -30,35 +33,34 @@ const defaultProps = {
 
 let Divider = (props) => {
   const {
-    inset,
-    muiTheme,
-    style,
+    _styleRoot,
     ...other,
   } = props;
 
-  const {
-    prepareStyles,
-  } = muiTheme;
-
-  const styles = {
-    root: {
-      margin: 0,
-      marginTop: -1,
-      marginLeft: inset ? 72 : 0,
-      height: 1,
-      border: 'none',
-      backgroundColor: muiTheme.rawTheme.palette.borderColor,
-    },
-  };
-
   return (
-    <hr {...other} style={prepareStyles(Object.assign({}, styles.root, style))} />
+    <hr {...other} style={_styleRoot} />
   );
 };
 
 Divider.displayName = 'Divider';
 Divider.propTypes = propTypes;
 Divider.defaultProps = defaultProps;
+
+const rootStyleSelector = createSelector('_styleRoot', [
```

Comment:
Hmmmm that could be interesting... Maybe if we name every style then we can forcefully provide overrides for every style object! let me investigate further.